### PR TITLE
Code clean up

### PR DESCRIPTION
**Pull request by LeAnthony**
- Refactor code to be more MVP
- Remove api gateway and common utils libraries  
- Added data layer (Retrofit & gson deserializer )
- Refactor layout.
   - extractor view 
   - Created base orientation layout with constraints view  
- Clean unused code 
- Update resource folder
   - style/theme
   - Dimension 
   - Color 
   - String 
   - Drawable (Selector)
* Not yet done 
- Clean up focus code 
- toolbar layout
- navigation layout

Full Screen 
Screen Shot 2019-05-08 at 1 27 42 AM
Split Screen 
Screen Shot 2019-05-08 at 1 30 10 AM

Quad screen 
Screen Shot 2019-05-08 at 1 28 15 AM

### CODE REVIEW

**Comment by LeAnthony** [comment]
Added Jitpack GMS dependency

File: app/src/main/java/com/tarsus/celebrity/view_layer/celeb/CelebrityPresenter.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            @Override
+            public void onFailure(Call<CelebrityPage> call, Throwable t) {
+                //TODO Error handling
+                t.printStackTrace();
+            }
+        });
+    }
+
+    public void requestCelebrityInfo(String celebrityId){
+        if(celebrityId == null || celebrityId.isEmpty() || this.celebrityId.equals(celebrityId)){
```
Comment (Comatose16) [inline app/src/main/java/com/tarsus/celebrity/view_layer/celeb/CelebrityPresenter.java]: Do we want to use a StringUtils library to check for null and blank?

File: app/src/main/java/com/tarsus/celebrity/view_layer/widget/PersistentFocusWrapper.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import android.widget.FrameLayout;
+
+import java.util.ArrayList;
+
+/**
+ * Saves the focused grandchild position.
+ * Helps add persistent focus feature to various ViewGroups.
+ * @hide
+ */
+public class PersistentFocusWrapper extends FrameLayout {
```
Comment (Comatose16) [inline app/src/main/java/com/tarsus/celebrity/view_layer/widget/PersistentFocusWrapper.java]: Did you take this out of the Contract? Can we pull all focus related methods in here?

File: app/src/main/java/com/tarsus/celebrity/view_layer/widget/SnapToStartLinearLayoutManager.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package com.tarsus.celebrity.view_layer.widget;
+
+import android.content.Context;
+import android.graphics.PointF;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.LinearSmoothScroller;
+import android.support.v7.widget.RecyclerView;
+
+public class SnapToStartLinearLayoutManager extends LinearLayoutManager {
```
Comment (Comatose16) [inline app/src/main/java/com/tarsus/celebrity/view_layer/widget/SnapToStartLinearLayoutManager.java]: What is the difference between this and ConstraintLayout? Is this inside the ConstraintLayout?

File: .idea/gradle.xml [hunk 4]
```diff
@@ -3,11 +3,16 @@
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
-        <compositeConfiguration>
-          <compositeBuild compositeDefinitionSource="SCRIPT" />
-        </compositeConfiguration>
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
+        <option name="modules">
```
Comment (Comatose16) [inline .idea/gradle.xml]: Should we check this file in? If not we can at it to the .gitignore

File: app/src/main/java/com/tarsus/celebrity/view_layer/ScreenDetectionUtil.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    public static String getCurrentScreenVariant(Activity activity){
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        activity.getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
+        int height = displayMetrics.heightPixels;
+        int width = displayMetrics.widthPixels;
+        int density = displayMetrics.densityDpi;
+
+
+        if((width/height) == (8/9)){
```
Comment (Comatose16) [inline app/src/main/java/com/tarsus/celebrity/view_layer/ScreenDetectionUtil.java]: Quad is when density is 1/3 of width.

**Pull request merged**